src/main/java/store/service/ProductServiceImpl.java

```diff
@@ -0,0 +1,126 @@
+package store.service;
+
+import store.exception.NotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import store.repository.ProductRepository;
+import store.model.Product;
+import store.request.ProductRequest;
+import store.request.UpdateProductRequest;
+
+import java.util.List;
+
+
+@Service
+public class ProductServiceImpl implements ProductService {
+    public ProductRepository productRepository;
+    ;
+ //   CategoryRepository categoryRepository;
+
+    @Autowired
+    ProductServiceImpl(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @Override
+    public Product getProductById(Long id) {
+        Product product;
+        try {
```

Comment:
its complicated for getting product. If you request product with wrong id - just return null